wit/guest.wit

```diff
@@ -1,10 +1,8 @@
-interface messaging-guest {
-    use messaging-types.{message, guest-configuration, error};
+interface guest {
+    use types.{message, config, error};
 
-    /// Returns the list of channels (and extension metadata within guest-configuration) that 
-    /// this component should subscribe to and be handled by the subsequent handler within guest-configuration
-    configure: func() -> result<guest-configuration, error>;
-
-    /// Whenever this guest receives a message in one of the subscribed channels, the message is sent to this handler
-    handler: func(ms: list<message>) -> result<_, error>;
+    /// Whenever this guest receives a message in one of the subscribed channels, the message is
+    /// sent to this handler. The guest is responsible for matching on the channel and handling the
+    /// message accordingly.
+    handler: func(ms: message) -> result<_, error>;
```

Comment:
Nit: to be symmetric with `wasi:http`, how about renaming the `guest` interface to `incoming-handler`, and renaming the `handler` function to `handle`?